src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -173,6 +214,21 @@ namespace ts.refactor.convertParamsToDestructuredObject {
             const symbol = checker.getSymbolAtLocation(node);
             return symbol && getSymbolTarget(symbol, checker);
         }
+
+        function getSymbolForInterfaceSignature(node: Node): Symbol | undefined {
+            const element = getContainingObjectLiteralElement(node);
+            if (element) {
+                const contextualType = element && checker.getContextualType(element.parent);
+                if (contextualType) {
+                    const name = getNameFromPropertyName(element.name);
+                    if (!name) return;
+                    if (!contextualType.isUnion()) {
+                        return contextualType.getProperty(name);
+                    }

```

Comment:
Seems like you probably need to do something more in case there's an intersection, or multiple overloads.

Reply:
Also, I can't remember if there's a way outside the checker to see that a symbol comes from a union or intersection, but it would be more elegant to get the contextual type of the property itself and then just check whether it's synthetic. Can you `getCheckFlags(symbol) & CheckFlags.Synthetic` outside the checker? There might be another way to do it.

Reply:
No longer offers the refactor for intersections or multiple signature overloads.